pkg/util/wimi/iswimi.go

```diff
@@ -0,0 +1,14 @@
+package wimi
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import "github.com/Azure/ARO-RP/pkg/api"
+
+// IsWimi checks whether a cluster is a Workload Identity cluster or Service Principal cluster
+func IsWimi(oc *api.OpenShiftCluster) bool {
+	if oc.Properties.PlatformWorkloadIdentityProfile != nil && oc.Properties.ServicePrincipalProfile == nil {
+		return true
+	}
+	return false
+}
```

Comment:
Following my first comment in the PR, if we want to have a richer domain and avoid an anemic domain model, we can move this function definition to pkg/api/openshiftcluster.go and have on line 29 something like:
```go
// IsWimi checks whether a cluster is a Workload Identity cluster or Service Principal cluster
func (oc *OpenShiftCluster) IsWimi() bool {
	if oc.Properties.PlatformWorkloadIdentityProfile != nil && oc.Properties.ServicePrincipalProfile == nil {
		return true
	}
	return false
}
```
which moves the business logic closer to where the data lives in 🙂

Reply:
One last minor comment about it, we can simplify the function if we change it to one single return where the true condition should be met 
```go
// IsWimi checks whether a cluster is a Workload Identity cluster or Service Principal cluster
func (oc *OpenShiftCluster) IsWimi() bool {
	return oc.Properties.PlatformWorkloadIdentityProfile != nil && oc.Properties.ServicePrincipalProfile == nil
}
```
Just mentioning it in case you find it easier to read.